Shopify/cli
# Fix functions multi save

### WHY are these changes introduced?
Function watching wasn't behaving as expected, producing multiple unnecessary builds in some cases and crashing on windows.


### WHAT is this pull request doing?
- Do not nullify the `buildController` on `finally`, because that AbortController is re-assigned every time there is a change, when we reach `finally` the `buildController` object is not the same that started the process.
- Don't throw when we fail to finish a subprocess. This throws on windows when a process finished just before we try to kill it, which will happen a lot of times. And there is no benefit on throwing here. If we really failed to kill a subprocess, throwing won't kill the subprocess either.


### How to test your changes?
- run `dev` having a function and unified deployments
- Open the function .js file and cmd+s multiple times
- You should see just 1 `Done!` 


### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
